department-of-veterans-affairs/abd-vro
# Update Frequently referenced github actions to use node20

This PR updates the two most frequently referenced github action job references to use the node20 upgrade in each of the job definitions.  For actions/checkout and actions/setup-java the default runtime for node will now use node20, rather then node16



## What was the problem?
When ever a github action used the checkout@v3 or setup-java@v3 job references, it would result in a warning about node 16 being deprecated. 

## How does this fix it?[^1]
By upgrading to use checkout@v4 and setup-java@v4 we'll be upgrading the actions to use node20 and we'll no longer have so many node 16 deprecation warnings.

## How to test this PR
- Observe fewer warnings in the action output of our workflows


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
